### ISSUE TITLE
Formatting sbt files formats metals.sbt even though it's in the .gitignore

### ISSUE THREAD

**Issue by ckipp01** [scalameta/sbt-scalafmt#154]
- **Configuration**:
```
version = "3.0.7"
project.git = true
```

## Problem

I'm not fully sure what changed, but as of the last release it seems that I can't get scalafmt to respect my gitignore settings to not format my `metals.sbt` settings. I have pretty much the same settings as I've always had, but now I'm always getting those file reported as formated like so:

```
[warn]   /Users/ckipp/Documents/scala-workspace/metals-minimal-sbt/project/metals.sbt
[warn]   /Users/ckipp/Documents/scala-workspace/metals-minimal-sbt/project/project/metals.sbt
[warn]   /Users/ckipp/Documents/scala-workspace/metals-minimal-sbt/project/project/project/metals.sbt
```

This happens when I run `scalafmtSbt` via sbt using the 2.4.3 version of the sbt plugin.

I have a minimal example project set up [here](https://github.com/ckipp01/metals-minimal-sbt) that you can just clone and use. Open it with metals, get your `metals.sbt` files generated, and then run `scalafmtSbt` and hopefully you'll see the same result as I'm seeing.

## Expectation

I'd expect this setting to respect my .gitconfig and not worry about my `metals.sbt` files.

The reason this is sort of a hassle is that when these are formatted and changed, metals detects that a changed happened in your build file which can cause Metals to prompt you to re-import your project.

**Comment by poslegm** [comment]
Hello! Last release has nothing to do with this issue, because sbt plugin doesn't support `project.git` at all. Related old issue: https://github.com/scalameta/sbt-scalafmt/issues/11. The trouble is that `project.git` code written in `scalafmt-cli` module, so it can't be used as dependency in this plugin (and other build tool integrations). The solutions requires a large refactoring and I haven't time to it in the near future 😞

**Comment by ckipp01** [comment]
> Hello! Last release has nothing to do with this issue, because sbt plugin doesn't support project.git at all.

Ahhh I didn't know this. 🤔 no idea why this is only popping up for me now this, as this should have happened to me all along. Any idea why the last release would have caused this to surface?

**Comment by poslegm** [comment]
I tried to downgrade scalafmt on your example project and receive same result 🤷🏻‍♂️

**Comment by kitbellew** [comment]
@poslegm this change might be helpful (to check `project.git`), along with the git functionality released in 3.1.0: scalameta/scalafmt#2880
